apis/kueue/v1beta1/localqueue_types.go

```diff
@@ -25,10 +25,13 @@ import (
 // LocalQueueSpec defines the desired state of LocalQueue
 type LocalQueueSpec struct {
 	// clusterQueue is a reference to a clusterQueue that backs this localQueue.
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="field is immutable"
 	ClusterQueue ClusterQueueReference `json:"clusterQueue,omitempty"`
 }
 
 // ClusterQueueReference is the name of the ClusterQueue.
+// +kubebuilder:validation:MaxLength=253
+// +kubebuilder:validation:Pattern="^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$"
```

Comment:
@jpbetz there is no CEL library for common k8s regexes, is there?

Reply:
Not yet, but @alexzielenski is working on adding them as [part of declarative validation](https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/4153-declarative-validation#formats), both OpenAPI formats and CEL utility functions to check common regexes (like dns1123label) are planned.